### PR TITLE
Project configuration

### LINKED ISSUE #8: Initial Project Configuration
**Describe the task that needs to be done.**

Before starting the implementation of our frontend, this issue requests the initialization of an empty React project.

**Describe how a solution to your proposed task might look like (and any alternatives considered).**

The project should also contain configurations for ESLint, Prettier, Jest, and React Testing Library. Also this is the proposed file structure that we should use. The project should also have easy-to-use scripts to perform linting.

```
.
└── root/
    └── components/
        ├── commons
        └── [page_name]/
            ├── [page_name.jsx]
            ├── [page_name.test.js]
            ├── [componentOne.jsx]
            └── [componentOne.test.js]
```

The `components` directory will contain all the React components used in the application. Within that directory, the `commons` directory will have (reusable) components which are used in multiple places, e.g. Navigation Menu, Buttons.

If components are specific to a particular page, you can create a directory for your page in `components` and place them there. Tests for all components will be stored along with the component file itself, as shown in the image above.

**Notes**

- The team has agreed on using vanilla JavaScript, and hence no TypeScript configuration would be required.

### PR DESCRIPTION
**Pull request by shreytailor**
## Description

The project is now initialized using `create-react-app`. It already comes with dependencies for React Testing Library / Jest, but I have done the setup for ESLint and Prettier. ESLint currently uses the default settings of airbnb, but we should be able to add more rules later if required.

I have also added in a script called `lint` which makes it easier to fix the linting issues in all of the files. To do this, you just invoke `npm run lint`.

Fixes #8

## How has this been tested?

N/A

## Screenshots

N/A

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# Getting Started with Create React App
```
[the remark this answers is not in the document]
Comment (shreytailor) [reply README.md:1]: Yeah I was thinking about just redoing the whole `README` file, but have we decided on a team/application name yet? Maybe for now we can just write `SOFTENG 701 Team 1`?

**Review by scorpionknifes — COMMENTED** [review]
Great work!

nit: 
Seems like `lockfileVersion: 2` 
Could we also set the min node version to 16 so we don't get people changing lockfile back to v1?

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
+
```
Comment (arathibiju) [inline .gitignore:2]: Looks good ! I'm not sure if we need all of it but I found a gitignore for react [here](https://github.com/facebook/react/blob/main/.gitignore) that had a bit more stuff. Especially the .vs code. Maybe worth adding that in?

**Review by arathibiju — APPROVED** [review]
Looks pretty good! Just added some small comments about the git-ignore and read me but overall looks great

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (arathibiju) [reply README.md:1]: Yeah that sounds good for now. If you can add links to the code of conduct and stuff it would be good.

**Review by arathibiju — APPROVED** [review]

**Review by scorpionknifes — APPROVED** [review]

**Pull request merged**